src/ir/named.rs

```diff
@@ -198,10 +199,24 @@ pub fn analyze<Analysis>(extra: Analysis::Extra) -> Analysis::Output
     let mut analysis = Analysis::new(extra);
     let mut worklist = analysis.initial_worklist();
 
+    // INVARIANT: node_ids_in_worklist.contains(node.id) if and only if
+    // worklist.contains(node)
+    let mut node_ids_in_worklist = BitSet::new();
+
     while let Some(node) = worklist.pop() {
+        // We just popped `node` from `worklist`, so remove `node` from
+        // `node_ids_in_worklist`
+        node_ids_in_worklist.remove(node.into());
```

Comment:
Can we `debug_assert!` that the node was in the worklist? It's good to do our best to make sure our invariants are upheld in the face of future changes.